arango/cursors/base.py

```diff
@@ -118,55 +123,28 @@ def next(self):
         :rtype: dict
         :raises: StopIteration, CursorNextError
         """
-        if not self.batch() and self.has_more():
-            res = self._conn.put("/_api/cursor/{}".format(self.id))
-            if res.status_code not in HTTP_OK:
-                raise CursorNextError(res)
-            self._data = res.body
-        elif not self.batch() and not self.has_more():
-            raise StopIteration
-        return self.batch().pop(0)
 
-    def close(self, ignore_missing=True):
-        """Close the cursor and free the resources tied to it.
+        if len(self.batch()) == 0:
+            if not self.has_more():
+                raise StopIteration
 
-        :returns: whether the cursor was closed successfully
-        :rtype: bool
-        :param ignore_missing: ignore missing cursors
-        :type ignore_missing: bool
-        :raises: CursorCloseError
-        """
-        if not self.id:
-            return False
-        res = self._conn.delete("/_api/cursor/{}".format(self.id))
-        if res.status_code not in HTTP_OK:
-            if res.status_code == 404 and ignore_missing:
-                return False
-            raise CursorCloseError(res)
-        return True
+            request = Request(
+                method="put",
+                url="/_api/{}/{}".format(self._cursor_type, self.id)
+            )
 
+            def handler(res):
+                if res.status_code not in HTTP_OK:
+                    raise CursorNextError(res)
+                return res.body
 
-class ExportCursor(Cursor):  # pragma: no cover
-    """ArangoDB cursor for export queries only.
+            job = self.handle_request(request, handler, job_class=BaseJob,
+                                      use_underlying=True)
```

Comment:
I still do not like the argument name `user_underlying` as its too vague and I cannot figure out what it does.

Reply:
use_underlying indicates whether the request should use the connection that created it or the underlying connection for that connection.  Since the underlying property recursively checks the underlying of its parents, this allows for things like having an asycnhronous execution wrapping a batch execution, while still allowing the AsyncJobs to use the underlying BaseConnection object to get their status, instead of using the batch connection

Reply:
If you have a better name for this behavior, let me know